src/Signature/SignatureV4.php

```diff
@@ -82,7 +82,9 @@ public function presign(
         $expires,
         array $options = []
     ) {
-        $startTimestamp = isset($options['start_time']) ? $this->convertToTimestamp($options['start_time']) : time();
+        $relativeTimeBase = isset($options['relativeTimeBase']) ? $options['relativeTimeBase'] : time();
+        $expiresTimestamp = $this->convertToTimestamp($expires,$relativeTimeBase);
```

Comment:
@kstich I think the segfault on the builds is coming from the precise to trusty upgrade, I can't replicate locally.  It also appears this is a change Travis made recently: https://blog.travis-ci.com/2017-07-11-trusty-as-default-linux-is-coming .  There were no segfaults before the upgrade, I can add the precise variable into the travis config if you want.  Let me know how you want to proceed.